utils/vector.hpp

```diff
@@ -350,11 +350,14 @@ const T* Vector<T>::end() const
 template <class T>
 void Vector<T>::erase(uint32_t position)
 {
-    for(uint32_t i = position; i < size; i++)
+    if(size != 0)
     {
-        array[i] = array [i + 1];
+        for(uint32_t i = position; i < size - 1; i++)
```

Comment:
This is so unnecessary, we can definitely do this in O(1)

Reply:
How are you going to move all the elements in O(1)?